### PR TITLE
Load history items from the database

### PR DESCRIPTION
**Pull request by pattmax00**

### CODE REVIEW

File: AcademicReward/Views/HistoryPage.xaml.cs [hunk 1]
```diff
@@ -9,33 +13,55 @@ namespace AcademicReward.Views;
...
+    // Load the history items from the database
+    public DatabaseErrorType LoadHistoryItems()
+    {
+        int userID = MauiProgram.Profile.ProfileID;
+
+        DatabaseErrorType dbError;
+        try
+        {
+            using var con = new NpgsqlConnection(Database.AcademicRewardsDatabase.InitializeConnectionString());
+            con.Open();
```
[the remark this answers is not in the document]
Comment (pattmax00) [reply AcademicReward/Views/HistoryPage.xaml.cs:35]: Fixing

File: AcademicReward/ModelClass/HistoryItem.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+{
+    /// <summary>
+    /// Primary Author: Maximilian Patterson
+    /// Secondary Author: None
+    /// Reviewer: 
+    /// </summary>
+    public class HistoryItem : ObservableObject
+    {
+        public int historyid { get; set; }
+        public int profileid { get; set; }
```
[the remark this answers is not in the document]
Comment (Wil-LaLonde) [reply AcademicReward/ModelClass/HistoryItem.cs:13]: All good, just trying to be consistent with C# standards.

**Comment by pattmax00** [comment]
Did a ton of refactoring and made the HistoryDatabase model. Interface now has a LoadItems() method which lets you give it an observable collection to update with columns loaded from the database. @Wil-LaLonde @Sean-Stille @xeelo2000

File: AcademicReward/Database/AcademicRewardsDatabase.cs [hunk 3]
```diff
@@ -1,12 +1,14 @@
 ﻿using Npgsql;
```
Comment (pattmax00) [inline AcademicReward/Database/AcademicRewardsDatabase.cs:1]: Changes here are just formatting

File: AcademicReward/Database/HistoryDatabase.cs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+            }
+            catch (NpgsqlException ex)
+            {
+                Console.WriteLine("Error while grabbing history: {0}", ex);
+                dbError = DatabaseErrorType.UsernameTakenDBError;
+            }
+            return dbError;
+        }
+
+        public DatabaseErrorType LookupItem(object obj)
```
Comment (pattmax00) [inline AcademicReward/Database/HistoryDatabase.cs:137]: LookupItem and LookupFullItem aren't really needed for this class

File: AcademicReward/Database/IDatabase.cs [hunk 5]
```diff
@@ -11,6 +13,7 @@ public interface IDatabase {
         public DatabaseErrorType UpdateItem(object obj);
         public DatabaseErrorType DeleteItem(object obj);
         public DatabaseErrorType LookupItem(object obj);
-        public DatabaseErrorType LookupFullItem(object obj); 
+        public DatabaseErrorType LookupFullItem(object obj);
+        public DatabaseErrorType LoadItems(ObservableCollection<object> obj, string[] args);
```
Comment (pattmax00) [inline AcademicReward/Database/IDatabase.cs:17]: Populates an observable collection with database data, good if you don't need a logic layer (like with History in my case)

File: AcademicReward/Database/LoginDatabase.cs [hunk 6]
```diff
@@ -165,5 +166,10 @@ public class LoginDatabase : AcademicRewardsDatabase, IDatabase {
             }
             return dbError;
         }
+
+        public DatabaseErrorType LoadItems(ObservableCollection<object> obj, string[] args)
```
Comment (pattmax00) [inline AcademicReward/Database/LoginDatabase.cs:170]: Needs to compile

File: AcademicReward/Views/HistoryPage.xaml.cs [hunk 7]
```diff
@@ -9,33 +14,16 @@ namespace AcademicReward.Views;
...
-		base.OnAppearing();
-
-		// Add some test data
-		HistoryItems.Add(new HistoryItem { Name = "Test 1", Description = "Description 1", Date = DateTime.Now });
-		HistoryItems.Add(new HistoryItem { Name = "Test 2", Description = "Description 2", Date = DateTime.Now });
-		HistoryItems.Add(new HistoryItem { Name = "Test 3", Description = "Description 3", Date = DateTime.Now });
-		HistoryItems.Add(new HistoryItem { Name = "Test 4", Description = "Description 4", Date = DateTime.Now });
-		HistoryItems.Add(new HistoryItem { Name = "Test 5", Description = "Description 5", Date = DateTime.Now });
-	}
+    IDatabase HistoryDatabase = new HistoryDatabase();
```
Comment (pattmax00) [inline AcademicReward/Views/HistoryPage.xaml.cs:17]: Just dump the data from this, no logic going on

**Review by xeelo2000 — APPROVED** [review]

**Review by xeelo2000 — APPROVED** [review]

**Review by xeelo2000 — APPROVED** [review]

**Review by xeelo2000 — APPROVED** [review]

**Pull request merged**